src/lsm/compaction.zig

```diff
@@ -1776,6 +1787,34 @@ pub fn CompactionType(
             beat.deactivate_assert_and_callback(.write, null);
         }
 
+        /// Return our blocks to the block pool.
+        pub fn beat_blocks_release(
```

Comment:
Ah yes, `release` is used in Grid terminology. `unassign` is good.

`bar_setup` is the odd one out; I was trying to go with more descriptive naming, but if you think it's clearer... (especially since some of the things have been overloaded, like `bar_setup_budget` sets the budget and some bar blocks)

Reply:
I don't feel strongly about the bar/beat terminology corresponding. But I think the before/after terminology should match in each case, e.g. assign/unassign, setup/teardown.